pkg/routeagent_driver/handlers/ovn/config_syncer.go

```diff
@@ -0,0 +1,118 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ovn
+
+import (
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner/pkg/routeagent_driver/constants"
+	"github.com/vishvananda/netlink"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+)
+
+func (ovn *Handler) startSyncer(stop chan struct{}) {
+	logger.Info("Starting route config syncer")
+	// Start monitoring routes
+	go ovn.monitorRoutingTable(stop)
+}
+
+func (ovn *Handler) monitorRoutingTable(stop chan struct{}) {
+	// Find the interface by name
+	iface, err := netlink.LinkByName(OVNK8sMgmntIntfName)
+	if err != nil {
+		logger.Errorf(err, "Failed to find interface: %q", OVNK8sMgmntIntfName)
+		return
+	}
+
+	// Start monitoring route deletion for the interface
+	routeCh := make(chan netlink.RouteUpdate)
+	doneCh := make(chan struct{})
+
+	err = netlink.RouteSubscribe(routeCh, doneCh)
+	if err != nil {
+		logger.Errorf(err, "failed to subscribe to route updates")
+		return
+	}
+
+	go func() {
+		for {
+			select {
+			case <-stop:
+				close(doneCh)
+				return
+			case update := <-routeCh:
+				// Check if the update is for the desired interface
+				if update.Route.LinkIndex == iface.Attrs().Index && update.Type == syscall.RTM_DELROUTE {
+					if err := ovn.handleRouteDeletion(); err != nil {
+						logger.Errorf(err, "error handling route syncing after %q router delete on interface after retries", OVNK8sMgmntIntfName)
+					} else {
+						logger.Infof("Successfully synced route after %q router delete on interface after retries", OVNK8sMgmntIntfName)
+					}
+				}
+			}
+		}
+	}()
+
+	<-stop
+}
+
+func (ovn *Handler) handleRouteDeletion() error {
+	backoff := wait.Backoff{
+		Cap:      6 * time.Minute,
```

Comment:
Since there is no `Jitter`, the duration between the retries is deterministic. Shall we keep the `Cap` as 3 minutes (instead of 6)?

Reply:
Done.